cupy/_creation/ranges.py

```diff
@@ -254,6 +254,104 @@ def logspace(start, stop, num=50, endpoint=True, base=10.0, dtype=None,
     return _core.power(base, y).astype(dtype)
 
 
+def _geomspace_scalar(start, stop, num=50, endpoint=True, dtype=None):
+    """
+    Scalar implementation for geomspace using NumPy for calculation.
+    Assumes start and stop are Python scalars.
+
+    Args:
+        start: Start of the interval.
+        stop: End of the interval.
+        num: Number of elements.
+        endpoint (bool): If ``True``, the stop value is included as the last
+            element. Otherwise, the stop value is omitted.
+        dtype: Data type specifier. It is inferred from the start and stop
+            arguments by default.
+    Returns:
+        cupy.ndarray: The 1-D array of ranged values.
+
+    .. seealso:: :func:`numpy.geomspace`
+    """
+    if start == 0 or stop == 0:
+        raise ValueError('Geometric sequence cannot include zero')
+
+    np_result = numpy.geomspace(start, stop, int(
+        num), endpoint=endpoint, dtype=dtype)
```

Comment:
Could you implement the `numpy.geomspace` equivalent for scalar, like done in https://github.com/cupy/cupy/blob/37d412f2df6b67f1bbe9923c497bcf58b812bf45/cupy/_creation/ranges.py#L64-L119? I think most of the code can be copied from [NumPy](https://github.com/numpy/numpy/blob/v2.1.0/numpy/_core/function_base.py#L323-L470)'s implementation but calling `cupy.logspace`.